nucleus/admin/monitor/src/main/java/fish/payara/admin/monitor/cli/SetMonitoringConfiguration.java

```diff
@@ -0,0 +1,259 @@
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.CommandLock;
+import org.glassfish.api.admin.CommandRunner;
+import org.glassfish.api.admin.CommandRunner.CommandInvocation;
+import org.glassfish.api.admin.ExecuteOn;
+import org.glassfish.api.admin.ParameterMap;
+import org.glassfish.api.admin.RestEndpoint;
+import org.glassfish.api.admin.RestEndpoints;
+import org.glassfish.api.admin.RuntimeType;
+import org.glassfish.config.support.CommandTarget;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.internal.api.Target;
+import org.jvnet.hk2.annotations.Service;
+import org.jvnet.hk2.config.ConfigSupport;
+import org.jvnet.hk2.config.TransactionFailure;
+
+import static org.glassfish.api.ActionReport.ExitCode.FAILURE;
+import static org.glassfish.api.ActionReport.ExitCode.SUCCESS;
+import static org.glassfish.api.ActionReport.ExitCode.WARNING;
+
+/**
+ * @author Susan Rai
+ */
+@Service(name = "set-monitoring-configuration")
+@PerLookup
+@CommandLock(CommandLock.LockType.NONE)
+@I18n("set.monitoring.configuration")
+@ExecuteOn(RuntimeType.DAS)
+@TargetType(value = {CommandTarget.DAS, CommandTarget.STANDALONE_INSTANCE, CommandTarget.CLUSTER, CommandTarget.CLUSTERED_INSTANCE, CommandTarget.CONFIG, CommandTarget.DEPLOYMENT_GROUP})
+@RestEndpoints({
+    @RestEndpoint(configBean = Domain.class,
+            opType = RestEndpoint.OpType.POST,
+            path = "set-monitoring-configuration",
+            description = "This command can be used to configure all the monitoring related configuration")
+})
+public class SetMonitoringConfiguration implements AdminCommand {
+
+    private static final Logger logger = Logger.getLogger(SetMonitoringConfiguration.class.getName());
+
+    @Inject
+    private Target targetUtil;
+
+    @Param(optional = true, alias = "monitoringEnabled")
+    private Boolean monitoringServiceEnabled;
+
+    @Param(optional = true, alias = "mbeanEnabled")
+    private Boolean mbeansEnabled;
+
+    @Param(optional = true)
+    private Boolean dTraceEnabled;
+
+    @Param(optional = true)
+    private Boolean amxEnabled;
+
+    @Param(optional = true)
+    private Boolean jmxLogEnabled;
+
+    @Param(optional = true)
+    private String jmxLogFrequency;
+
+    @Param(optional = true, acceptableValues = "NANOSECONDS,MILLISECONDS,SECONDS,MINUTES,HOURS,DAYS")
+    private String jmxLogFrequencyUnit;
+
+    @Param(name = "addattribute", optional = true, alias = "addproperty")
+    private String attributesToAdd;
+
+    @Param(name = "delattribute", optional = true, alias = "delproperty")
+    private String attributesToRemove;
+
+    @Param(optional = true)
+    private Boolean restMonitoringEnabled;
+
+    @Param(optional = true)
+    private String applicationName;
+
+    @Param(optional = true)
+    private String contextRoot;
+
+    @Param(optional = true)
+    private Boolean securityEnabled;
```

Comment:
It might be worth prepending these with `restmonitoring` to highlight them as being specific to the rest monitoring.
So:
* `--restmonitoringenabled`
* `--restmonitoringapplicationname`
* `--restmonitoringcontextroot`
* `--restmonitoringsecurityenabled`